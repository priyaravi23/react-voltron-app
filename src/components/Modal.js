import React from 'react';
import {oneOf, arrayOf, string, func, object} from 'prop-types';

import './Modal.scss';
import Button from "./Button";
import {EMPTY_FUNC} from "../utils/utils";

const Modal = (props) => {
    const {title, className, onClose = EMPTY_FUNC, actionButtons = null, errMessage} = props;
    return (
        <div className="modal-backdrop">
            <section className={`modal ${className}`}>
                {title && <h2 className={'title'}>{title}</h2>}
                <div className="modal-body">
                    {props.children}
                </div>
                <div className="action-buttons">
                    {errMessage && <div className="error-message">{errMessage}</div>}
                    <Button text={'close'} onClick={onClose}/>
                    {actionButtons}
                </div>
            </section>
        </div>
    );
};

Modal.propTypes = {
    title: string,
    onClose: func,
    actionButtons: arrayOf(object),
    errMessage: string
};

export default Modal;