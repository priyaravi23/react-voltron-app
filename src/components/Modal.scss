.modal-backdrop {
  height: 100vh;
  width: 100vw;
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0,0,0,0.4);
}

.modal {
  position:fixed;
  background: white;
  width: 80%;
  top:50%;
  left:50%;
  transform: translate(-50%,-50%);
  padding: 2em;
  border-radius: 0.5em;
  > .title {
    text-transform: capitalize;
    font-weight: 300;
  }
  .action-buttons {
    padding: 1em;
    display: flex;
    align-items: center;
    justify-content: flex-end;
    > .btn {
      margin-left: 1em;
    }
  }
}

@supports (backdrop-filter: blur(3px)) {
  .modal-backdrop {
    background-color: transparent;
    backdrop-filter: blur(20px);
  }
}