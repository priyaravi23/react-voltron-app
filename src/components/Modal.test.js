import React from 'react'
import Modal from './Modal';
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() });

describe('<Modal /> shallow rendering', () => {
    it('h2 contains correct text', () => {
        const wrapper = shallow(<Modal title={'Failing Stars Modal'} />);
        expect(wrapper.find('h2').text()).toBe('Failing Stars Modal');
    });

    it('matches the snapshot', () => {
        const tree = shallow(<Modal />);
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('on `close` button click close modal window', () => {
        const onClose = jest.fn();
        const wrapper = shallow(<Modal onClose={onClose}/>);
        const button = wrapper.find('.close');
        button.simulate('click');
        expect(onClose).toHaveBeenCalled();
    });
});

describe('<Modal /> mount rendering', () => {
    it('h2 contains correct text', () => {
        const wrapper = mount(<Modal title={'Failing Stars Modal'}/>);
        expect(wrapper.find('h2').text()).toBe('Failing Stars Modal');
        wrapper.unmount()
    });

    it('matches the snapshot', () => {
        const tree = mount(<Modal />);
        expect(toJson(tree)).toMatchSnapshot();
        tree.unmount()
    })
});