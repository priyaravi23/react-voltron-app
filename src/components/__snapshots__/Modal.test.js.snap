// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> mount rendering matches the snapshot 1`] = `
<Modal>
  <div
    className="modal-backdrop"
  >
    <section
      className="modal undefined"
    >
      <div
        className="modal-body"
      />
      <div
        className="action-buttons"
      >
        <Button
          className="close"
          onClick={[Function]}
          text="close"
        >
          <a
            className="btn close"
            href="#"
            onClick={[Function]}
          >
            close
          </a>
        </Button>
      </div>
    </section>
  </div>
</Modal>
`;

exports[`<Modal /> shallow rendering matches the snapshot 1`] = `
<div
  className="modal-backdrop"
>
  <section
    className="modal undefined"
  >
    <div
      className="modal-body"
    />
    <div
      className="action-buttons"
    >
      <Button
        className="close"
        onClick={[Function]}
        text="close"
      />
    </div>
  </section>
</div>
`;
